### PR TITLE
Update dependencies, simplify script plugins

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR:
  * Bumps versions of Kover, detekt, and ErrorProne Gradle plugin.
  * Merges `java-module` script plugin into `kotlin-jvm-module.

### CODE REVIEW

File: buildSrc/src/main/kotlin/jacoco-kmm-jvm.gradle.kts
Comment (yevhenii-nadtochii) [inline buildSrc/src/main/kotlin/jacoco-kmm-jvm.gradle.kts:1]: Should we use KMM abbreviation? It stands for Kotlin Multiplatform Mobile. 

We could go with KM or KMP.

File: buildSrc/src/main/kotlin/kotlin-jvm-module.gradle.kts
Comment (yevhenii-nadtochii) [inline buildSrc/src/main/kotlin/kotlin-jvm-module.gradle.kts:1]: Or simpler `jvm-module`. And think of Spine JVM module as one that can be Java or Kotlin, or mixed.

Comment (alexander-yevsyukov) [reply buildSrc/src/main/kotlin/kotlin-jvm-module.gradle.kts:1]: Let's go with `jvm-module`. I'll rename.

File: buildSrc/src/main/kotlin/jacoco-kmm-jvm.gradle.kts
Comment (alexander-yevsyukov) [reply buildSrc/src/main/kotlin/jacoco-kmm-jvm.gradle.kts:1]: I've made it simply `jacoco-kotlin-jvm` because it would tell directly what it's for.

**Review by yevhenii-nadtochii — APPROVED** [review]

**Pull request merged**